### PR TITLE
ACL fix for SuggestAttributeSets.php

### PR DESCRIPTION
**Pull request by rossluk**
Currently if you create custom role with full access ("Resource Access - Custom" with all checkboxes checked) to admin this role doesn't have access to load attribute sets on /admin/catalog/product/new. "Resource Access - All" works well. I've added the inheritance from \Magento\Catalog\Controller\Adminhtml\Product like in SuggestAttributes.php for right _isAllowed() acl check.

### CODE REVIEW

**Comment by davidalger** [comment]
@rossluk Thank you for your contribution. Please accept the contributors license agreement so this PR can be further processed. This can be done by clicking the "Details" link next to the "license/cla" check below.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/SuggestAttributeSets.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
  */
 namespace Magento\Catalog\Controller\Adminhtml\Product;
 
-class SuggestAttributeSets extends \Magento\Backend\App\Action
+class SuggestAttributeSets extends \Magento\Catalog\Controller\Adminhtml\Product
```
Comment (antonkril) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/SuggestAttributeSets.php]: Inheritance introduces hard-coded dependency and unused behavior. Please use specific ACL resource for this action.

Comment (rossluk) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/SuggestAttributeSets.php]: I thought that but I was confused by logic in SuggestAttributes.php. Now I've checked how it works and refactored both files.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/SuggestAttributeSets.php [hunk 2]
```diff
@@ -46,4 +46,15 @@ public function execute()
         );
         return $resultJson;
     }
+    
+    /**
+     * Check for is allowed
+     *
+     * @return boolean
+     */
+    protected function _isAllowed()
```
Comment (vkublytskyi) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/SuggestAttributeSets.php]: No need to override Magento\Backend\App\AbstractAction::_isAllowed() as it uses Late Static Binding. More accurate way is to define ADMIN_RESOURCE constant.

Comment (rossluk) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/SuggestAttributeSets.php]: You are right but why this logic isn't used nowhere in project?
For instance:
Magento_Catalog/Controller/Adminhtml/Category.php
Magento_Catalog/Controller/Adminhtml/Product.php
Magento_Cms/Controller/Adminhtml/Block.php
Magento_Sales/Controller/Adminhtml/Order.php
etc.

Comment (vkublytskyi) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/SuggestAttributeSets.php]: Listed files have been fixed already in https://github.com/magento/magento2/commit/148bae13d759f133a793723ff9bcd2b55ac7ebc2.
Unfortunately we still have some legacy code that require refactoring and will appreciate your help in this.

Comment (rossluk) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/SuggestAttributeSets.php]: Thanks for you answer. I've updated the PR.

**Comment by vkublytskyi** [comment]
@rossluk Thanks for your help to make Magento better. Please sign Contributor License Agreement and we will merge this PR

**Comment by yaronish** [comment]
@rossluk Thanks for your help to make Magento better. Please sign Contributor License Agreement and we will merge this PR

**Comment by rossluk** [comment]
@yaronish I've done it many times but it still not signed.

**Comment by vkublytskyi** [comment]
@rossluk Sorry for an issue with CLA. We will merge your PR. One more time thanks for your contribution and for your patience

**Pull request merged**